tests/integration/cos_substrate.py

```diff
@@ -0,0 +1,261 @@
+        """Create a COS substrate.
+
+        Returns:
+            str: The generated kubeconfig.
+        """
+        ...
+
+    def teardown_substrate(self): 
+        """Teardown the COS substrate."""
+        ...
+
+
+class LXDSubstrate(COSSubstrate):
+    """A COS substrate implemented using LXD."""
+
+    def __init__(self, container_name: str, network_name: str) -> None:
+        """Initialize LXDSubstrate instance.
+
+        Args:
+            container_name (str): Name of the container.
+            network_name (str): Name of the network.
+        """
+        self.client = Client()
+        self.container_name = container_name
+        self.network_name = network_name
+
+    def apply_profile(self, profile_name="microk8s.profile", target_profile_name="cos-profile"):
+        """Apply LXD profile.
+
+        Args:
+            profile_name (Optional[str]): Name of the profile to apply.
+            target_profile_name (Optional[str]): Name of the target profile. Defaults to 'cos-profile'.
+        """
+        profile_path = Path("tests/integration/data") / profile_name
+        with open(profile_path) as file:
+            try:
+                raw_profile = yaml.safe_load(file)
+                config = raw_profile.get("config", {})
+                devices = raw_profile.get("devices", {})
+                self.client.profiles.create(target_profile_name, config=config, devices=devices)
+                log.info(f"Profile {target_profile_name} applied successfully.")
+            except (yaml.YAMLError, Exception) as e:
+                log.error(f"Failed to read or apply LXD profile: {e}")
+
+    def create_container(self, name: str):
+        """Create a container.
+
+        Args:
+            name (str): Name of the container.
+
+        Returns:
+            container: The created container instance, or None if creation fails.
+        """
+        log.info("Creating container")
+        config = {
+            "name": name,
+            "source": {
+                "type": "image",
+                "mode": "pull",
+                "server": "https://cloud-images.ubuntu.com/releases",
+                "protocol": "simplestreams",
+                "alias": "22.04",
+            },
+            "type": "container",
+            "devices": {},
+            "profiles": ["default", "cos-profile"],
+        }
+        if self.network_name:
+            config["devices"]["eth0"] = {
+                "name": "eth0",
+                "nictype": "bridged",
+                "parent": self.network_name,
+                "type": "nic",
+            }
+        try:
+            container = self.client.instances.create(config, wait=True)
+            log.info("Starting Container")
+            container.start(wait=True)
+            time.sleep(60)
+            return container
+
+        except Exception as e:
+            log.error(f"Failed to create or start container: {e}")
+            return None
+
```

Comment:
alright -- but is there any chance we can just use our current juju to create another model for microk8s?  Even if that model is holding a bare machine (not necessarily the microk8s charm).


models in this controller would be:
1) controller model
2) test_k8s-1234 <-- model under test
3) k8s-holding-cos-1234 <-- model holding the kubernetes machine used for cos
    4) cos-1234 <--model atop the `add_k8s` from `#3`